torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -47,18 +58,49 @@ def regenerate_view(self, bases_list: List[Tensor]):
         if self.is_alias:
             return torch.ops.aten.alias.default(base)
 
-        assert self.stride is not None
-        assert self.size is not None
-        assert self.storage_offset is not None
+        if self.slice_info is not None:
+            return torch.ops.aten.slice.Tensor(
+                bases_list[self.base_index], *self.slice_info
+            )
 
+        assert self.as_strided_info is not None
         return torch.as_strided(
-            base,
-            self.size,
-            self.stride,
-            self.storage_offset,
+            bases_list[self.base_index],
+            *self.as_strided_info,
         )
 
 
+# return None or (dim, start, end)
+def try_use_slice(base, tensor):
+    if (
+        base.storage_offset() == tensor.storage_offset()
+        and base.stride() == tensor.stride()
+        and base.size() == tensor.size()
+    ):
+        return (0, 0, base.size()[0])
+
+    # TODO can we relax this for better coverage?
```

Comment:
what do you mean?

Reply:
does slice always preserve stride() and size()? I though not. 
I mean there might be other cases where we can regenerate slice

Reply:
slice() semantics are that it always returns a tensors with the same strides and it returns a tensor with the same sizes in all but one dimension. If we're talking about reconstructing the result of 2+ slices then... we're looking at auto_functionalized_v3